sql/log_event.cc

```diff
@@ -1932,7 +1934,7 @@ Query_log_event::begin_event(String *packet, ulong ev_offset,
   int2store(p + FLAGS_OFFSET, flags);
 
   p[EVENT_TYPE_OFFSET]= QUERY_EVENT;
-  int4store(q + Q_THREAD_ID_OFFSET, 0);
+  int4store(q + Q_THREAD_ID_OFFSET, original_event.thread_id);
```

Comment:
Let's add an extra condition here, to ensure `original_events.flags_extra & FL_EXTRA_THREAD_ID` must be set, in order to use `original_event.thread_id`, and otherwise `0`. Where the current state of the code _should_ guarantee that, let's make this check just in case, as we don't have any MTR testing of this, and we don't want this to silently break from future changes.

This is so we don't overwrite the `thread_id` for events that originate before MDEV-7850, e.g. from old binary logs, or in chain replication setups like `MariaDB 11.4` -> `MariaDB 11.5+` -> `MySQL 5.7`.